src/Node/Expression/InlinePrint.php

```diff
@@ -26,10 +26,19 @@ public function __construct(Node $node, int $lineno)
 
     public function compile(Compiler $compiler): void
     {
-        $compiler
-            ->raw('print (')
-            ->subcompile($this->getNode('node'))
-            ->raw(')')
-        ;
+        if ($compiler->getEnvironment()->useYield()) {
+            $compiler
+                ->raw('yield ')
```

Comment:
do we need parenthesis around the yielded expression as done for `print` ? Or are the precedence fine for yield ?

Reply:
We tend to over use parentheses in generated code.